cognite/client/data_classes/simulators/simulators.py

```diff
@@ -862,55 +862,158 @@ class SimulatorModel(CogniteResource):
-        self.created_time = created_time
-        self.last_updated_time = last_updated_time
         self.name = name
-        self.type_key = type_key
+        self.type = type
         self.description = description
 
     @classmethod
-    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+    def _load(
+        cls: type[T_SimulatorModel], resource: dict[str, Any], cognite_client: CogniteClient | None = None
+    ) -> T_SimulatorModel:
+        instance = super()._load(resource, cognite_client)
+        return instance
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return super().dump(camel_case=camel_case)
+
+
+T_SimulatorModel = TypeVar("T_SimulatorModel", bound=SimulatorModelCore)
+
+
+class SimulatorModelWrite(SimulatorModelCore):
+    def __init__(
+        self,
+        external_id: str | None = None,
+        simulator_external_id: str | None = None,
+        data_set_id: int | None = None,
+        name: str | None = None,
+        type: str | None = None,
+        description: str | None = None,
+    ) -> None:
+        super().__init__(
+            external_id=external_id,
+            simulator_external_id=simulator_external_id,
+            data_set_id=data_set_id,
+            name=name,
+            type=type,
+            description=description,
+        )
+
+    @classmethod
+    def _load(cls, resource: dict, cognite_client: CogniteClient | None = None) -> SimulatorModelWrite:
         return cls(
-            id=resource["id"],
             external_id=resource["externalId"],
             simulator_external_id=resource["simulatorExternalId"],
-            name=resource["name"],
             data_set_id=resource["dataSetId"],
-            created_time=resource["createdTime"],
-            last_updated_time=resource["lastUpdatedTime"],
-            type_key=resource.get("typeKey"),
+            name=resource["name"],
+            type=resource.get("typeKey"),
             description=resource.get("description"),
         )
 
+    def as_write(self) -> SimulatorModelWrite:
+        """Returns self."""
+        return self
+
+
+class SimulatorModel(SimulatorModelCore):
+    """
+    The simulator model resource represents an asset modeled in a simulator.
+    This asset could range from a pump or well to a complete processing facility or refinery.
+    The simulator model is the root of its associated revisions, routines, runs, and results.
+    The dataset assigned to a model is inherited by its children. Deleting a model also deletes all its children, thereby
+    maintaining the integrity and hierarchy of the simulation data.
+
+    Simulator model revisions track changes and updates to a simulator model over time.
+    Each revision ensures that modifications to models are traceable and allows users to understand the evolution of a given model.
+
+    Limitations:
+        - A project can have a maximum of 1000 simulator models
```

Comment:
Not sure we should have this duplicated here as well. If we change it in the API spec in the future, we would need to hunt all the other mentions that could refer to an outdated limit.

Reply:
agree, also those are generated from the actual values we use in the code, so it's better to have a single source of truth as we have now